smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/waiters/WaiterAcceptorGenerator.kt

```diff
@@ -58,7 +58,7 @@ class WaiterAcceptorGenerator(
                 }
                 is Matcher.ErrorTypeMember -> {
                     writer.write("guard case .failure(let error) = result else { return false }")
-                    writer.write("return (error as? WaiterTypedError)?.waiterErrorType == \$S", matcher.value)
+                    writer.write("return (error as? ServiceError)?.typeName == \$S", matcher.value)
```

Comment:
Using `ServiceError` instead of `WaiterTypedError` since the waiter-specific protocol is eliminated.